packages/notification-center/src/components/novu-provider/NovuProvider.tsx

```diff
@@ -73,47 +67,39 @@ export function NovuProvider(props: INovuProviderProps) {
 }
 
 interface ISessionInitializationProps {
-  applicationIdentifier: string;
-  subscriberId?: string;
+  onInit: (flag: boolean) => void;
   children: JSX.Element;
 }
 
-function SessionInitialization({ children, ...props }: ISessionInitializationProps) {
-  const { api: apiService } = useApi();
+function SessionInitialization({ onInit, children }: ISessionInitializationProps) {
+  const { api } = useApi();
   const { applyToken, setUser } = useAuth();
-  const { onLoad, subscriberHash } = useContext<INovuProviderContext>(NovuContext);
-
-  useEffect(() => {
-    if (props?.subscriberId && props?.applicationIdentifier) {
-      (async (): Promise<void> => {
-        await initSession({
-          clientId: props.applicationIdentifier,
-          data: { subscriberId: props.subscriberId },
-          subscriberHash,
-        });
-      })();
-    }
-  }, [props?.subscriberId, props?.applicationIdentifier]);
+  const { applicationIdentifier, subscriberId, subscriberHash, onLoad } = useContext<INovuProviderContext>(NovuContext);
 
-  async function initSession(payload: { clientId: string; data: { subscriberId: string }; subscriberHash: string }) {
+  const initSession = useCallback(async () => {
     if ('parentIFrame' in window) {
       // eslint-disable-next-line @typescript-eslint/no-explicit-any
       (window as any).parentIFrame.autoResize(true);
     }
 
-    const response = await apiService.initializeSession(payload.clientId, payload.data.subscriberId, subscriberHash);
-
-    api.setAuthorizationToken(response.token);
```

Comment:
This method will call inside `applyToken`, so there isn't necessary to call it separately